src/oscar/templatetags/category_tags.py

```diff
@@ -3,10 +3,50 @@
 from oscar.core.loading import get_model
 
 register = template.Library()
-Category = get_model('catalogue', 'category')
+Category = get_model("catalogue", "category")
 
 
-@register.simple_tag(name="category_tree")
+class CheapCategoryInfo(dict):
+    """
+    Wrapper class for Category.
+
+    Besides allowing inclusion of extra info, useful while rendering a template,
+    this class hides any expensive properties people should not use by accident
+    in templates.
+
+    This replaces both the node as the info object returned by the ``category_tree``
+    templatetag, so it mimics a tuple of 2 items (which are the same) for
+    backwards compatibility.
+    """
+
+    def __init__(self, category, **info):
+        super(CheapCategoryInfo, self).__init__(info)
+        self.category = category
+        self.field_names = set(
+            [field.name for field in category._meta.get_fields(include_parents=True)]
+        )
+
```

Comment:
This + the `__getattr__` stuff can be replaced by setting the data up front.

```python
fieldnames = set([field.name for field in category._meta.get_fields(include_parents=True)])
self.__dict__.update(dict([(name, getattr(category, name)) for name in fieldnames)]))
```
Or 
```python
fieldnames = set([field.name for field in category._meta.get_fields(include_parents=True)])
info.update(dict([(name, getattr(category, name)) for name in fieldnames)]))
super(CheapCategoryInfo, self).__init__(info)
```

This allows for faster access by skipping the `__getattr__` call.

Reply:
done with a metaclass